prediction_market_agent/agents/crewai_subsequential_agent/crewai_agent_subquestions.py

```diff
@@ -0,0 +1,192 @@
+import typing as t
+
+from crewai import Agent, Crew, Process, Task
+from langchain_core.language_models import BaseChatModel
+from langchain_openai import ChatOpenAI
+from pydantic import BaseModel
+
+from prediction_market_agent.agents.crewai_subsequential_agent.prompts import (
+    CREATE_OUTCOMES_FROM_SCENARIO_OUTPUT,
+    CREATE_OUTCOMES_FROM_SCENARIO_PROMPT,
+    FINAL_DECISION_PROMPT,
+    PROBABILITY_CLASS_OUTPUT,
+    PROBABILITY_FOR_ONE_OUTCOME_PROMPT,
+    RESEARCH_OUTCOME_OUTPUT,
+    RESEARCH_OUTCOME_PROMPT,
+)
+from prediction_market_agent.tools.crewai_tools import TavilyDevTool
+from prediction_market_agent.utils import APIKeys
+
+tavily_search = TavilyDevTool()
+
+
+class Outcomes(BaseModel):
+    outcomes: list[str]
+
+
+class ProbabilityOutput(BaseModel):
+    decision: str
+    p_yes: float
+    p_no: float
+    confidence: float
+
+
+class CrewAIAgentSubquestions:
+    def __init__(self) -> None:
+        llm = self._build_llm()
+        self.researcher = Agent(
+            role="Research Analyst",
+            goal="Research and report on some future event, giving high quality and nuanced analysis",
+            backstory="You are a senior research analyst who is adept at researching and reporting on future events.",
+            verbose=True,
+            allow_delegation=False,
+            tools=[tavily_search],
+            llm=llm,
+        )
+
+        self.predictor = Agent(
+            role="Professional Gambler",
+            goal="Predict, based on some research you are presented with, whether or not a given event will occur",
+            backstory="You are a professional gambler who is adept at predicting and betting on the outcomes of future events.",
+            verbose=True,
+            allow_delegation=False,
+            llm=llm,
+        )
+
+    def _build_llm(self) -> BaseChatModel:
+        keys = APIKeys()
+        llm = ChatOpenAI(
+            model="gpt-3.5-turbo-0125",
+            openai_api_key=keys.openai_api_key.get_secret_value(),  # type: ignore
```

Comment:
Why does it need type ignore?

Reply:
![image](https://github.com/gnosis/prediction-market-agent/assets/4126235/0e3736ea-ae26-414c-91f9-eac2569b10a3)

Reply:
But it's explicitly stated liked this in the Langchain docs -> https://python.langchain.com/docs/integrations/chat/openai/

Reply:
It's possible to click through the ChatOpenAI code and check out what's the problem.

In this case is because mypy expects the aliased field for some reason:

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/2ead24d5-a02a-466e-a400-54050fd625a8)

```suggestion
            api_key=keys.openai_api_key.get_secret_value(),
```

This works

Reply:
great catch!